voila/handler.py

```diff
@@ -117,8 +117,19 @@ def redirect_to_file(self, path):
     @tornado.gen.coroutine
     def _jinja_kernel_start(self):
         assert not self.kernel_started, "kernel was already started"
+
+        # Adding request uri to kernel env
+        kernel_env = os.environ.copy()
+        kernel_env['REQUEST_URI'] = str(self.request.uri)
```

Comment:
I don't think so, I imagine that for many languages there are CGI tools, and they would look for this env var. This would for instance work with php kernels :)
There must also be a ton of perl, c, c++ libraries that know how to handle it and I wonder if we should include more variables.

Reply:
And if we want to stay close to CGI, shouldn't `QUERY_STRING` also be passed?

Reply:
Yah, I like this. `REQUEST_URI` and `QUERY_STRING` are a good place to start. Any other properties that we should add for this to be minimally compliant/usable with CGI tools?

Reply:
> But we have to be careful with the variables we want to include to not leak sensitive information.

I think now the environment is passed already, could this include sensitive information, like tokens of non-notebook authors. This is a separate issue for which we may open another issue?

Reply:
Yes, currently the full environment is passed (we would just be adding to it here). I would be cautious about removing the environment. There are some common setups (like the Google Cloud SDK) that store values in the environment. People are using and depending on these in their Jupyter setup and it would feel strange if Voila didn't match the same behaviour. A separate conversation though :)

Reply:
Agree that it can be a separate conversation :+1: